src/coprocessor/statistics/analyze.rs

```diff
@@ -51,15 +49,31 @@ impl<'a> AnalyzeContext<'a> {
         AnalyzeContext {
             req: req,
             snap: snap,
-            statistics: statistics,
             ranges: ranges,
         }
     }
 
-    pub fn handle_request(self) -> Result<Response> {
+    pub fn handle_request(mut self, statistics: &mut Statistics) -> Result<Response> {
         let ret = match self.req.get_tp() {
-            AnalyzeType::TypeIndex => self.handle_index(),
-            AnalyzeType::TypeColumn => self.handle_column(),
+            AnalyzeType::TypeIndex => {
+                let req = self.req.take_idx_req();
```

Comment:
Why not keep them in `handle_index` just like the old way?

Reply:
Because we should collect statistics from scanner when error happens?